pkg/apis/flowcontrol/validation/validation.go

```diff
@@ -332,6 +342,42 @@ func ValidatePriorityLevelConfigurationCondition(condition *flowcontrol.Priority
 	return allErrs
 }
 
+// ValidateNonResourceURLPath validates non-resource-url path by following rules:
+//   1. Slash must be the leading character of the path
+//   2. A non-resource-url path must not end up with slash
+//   3. Continuous/double slash is forbidden in the path
+//   4. White-space is forbidden in the path
+//   5. Wildcard "*" should be only character in a segment if present
```

Comment:
what is the use-case for honoring embedded wildcards (as opposed to suffix-only wildcards)?

Reply:
that depends on whether wildcard can match slash or not. rbac matcher matches slash, should we align w/ rbac in the first place? 

https://github.com/kubernetes/kubernetes/blob/1488460cd2c4948da5fa613655b73e92ff788c96/pkg/apis/rbac/v1/evaluation_helpers.go#L107-L109

Reply:
i updated to align w/ rbac. the wildcard in rbac matches slash, e.g. `/healthz/*` will match `/healthz/poststarthook/start-kube-apiserver-informers`. if wildcard doesnt match slash, we have to set `/healthz/*/*` otherwise. i just added a note to emphasis that..